paddle/fluid/operators/pool_mkldnn_op.cc

```diff
@@ -161,9 +156,20 @@ class PoolMKLDNNOpKernel : public paddle::framework::OpKernel<T> {
       dev_ctx.SetBlob(key_pool_src_mem_p, src_memory);
       dev_ctx.SetBlob(key_pool_dst_mem_p, dst_memory);
 
-      pool_p = std::make_shared<pooling_forward>(*pool_pd, *(src_memory.get()),
-                                                 *(dst_memory.get()),
-                                                 *workspace_memory);
+      if (is_test) {
+        pool_p = std::make_shared<pooling_forward>(
+            *pool_pd, *(src_memory.get()), *(dst_memory.get()));
+      } else {
```

Comment:
Doesn't doing `*src_memory` have the same effect as `*(src_memory.get())`. If I'm not mistaken, `std::shared_ptr::operator*()` dereferences stored pointer, so it's the same as doing `*get()` on shared pointer.

Reply:
Ok, I changed that.